orc8r/cloud/go/services/analytics/calculations/calculations.go

```diff
@@ -131,26 +147,23 @@ func CombineLabels(l1, l2 map[string]string) map[string]string {
 	return retLabels
 }
 
-//RegisterResults exports the metrics to prometheus
-func RegisterResults(calc CalculationParams, results []*protos.CalculationResult) {
+// RegisterResults exports the metrics to prometheus
+func RegisterResults(calc *CalculationParams, results []*protos.CalculationResult) {
```

Comment:
Here and elsewhere: unless you have a specific reason, don't use pointers as elements of slices. Should instead be `[]protos.CalculationResult`

https://medium.com/@philpearl/bad-go-slices-of-pointers-ed3c06b8bb41

Reply:
Now I recall. The reason why I made it this way is because, the proto definition generates this code
```
// CollectResponse provides the response of the built analytics response
message CollectResponse {
  repeated CalculationResult results = 1;
}
// CollectResponse provides the response of the built analytics response
type CollectResponse struct {
	Results              []*CalculationResult `protobuf:"bytes,1,rep,name=results,proto3" json:"results,omitempty"`
	XXX_NoUnkeyedLiteral struct{}             `json:"-"`
	XXX_unrecognized     []byte               `json:"-"`
	XXX_sizecache        int32                `json:"-"`
}
```
I modified the code accordingly to pass in slice of pointers than slice of objects here

Reply:
True I wasn't paying attention to the fact that it was generated. I agree it's not worth dealing with, especially since we do use ptr slices for protos elsewhere 🙂